### PR TITLE
fixes several issues with update-entity w/ test cases

### LINKED ISSUE #529: Cannot read property 'user' of undefined - lib\socket\update-entity.js:26:104
Getting the following error:

error: Unhandled Rejection at: Promise  Promise {
  <rejected> TypeError: Cannot read property 'user' of undefined
    at `local-path`\node_modules\feathers-authentication\lib\socket\update-entity.js:26:104
    at Array.forEach (native)
    at Object.updateEntity (`local-path`\node_modules\feathers-authentication\lib\socket\update-entity.js:24:26)

Looks to be an issue when assigning socketEntity:

feathers-authentication/src/socket/update-entity.js
```JavaScript
Object.keys(socketMap).forEach(socketId => {
    const socket = socketMap[socketId];
    const socketEntity = (socket.feathers && socket.feathers[authConfig.entity]) || socket.request.feathers[authConfig.entity];
\\...
```
I would change that to:

```JavaScript
Object.keys(socketMap).forEach(socketId => {
    const socket = socketMap[socketId];
    const feathers = socket.feathers || socket.request.feathers;
    const socketEntity = feathers && feathers[authConfig.entity];
\\...
```
I'll try and submit a pull request... if I can find the time!

### PR DESCRIPTION
**Pull request by jerfowler**
Fixes #529 

 √ gracefully handles unauthenticated connections
 √ updates the passed-in entity when the idField is an ObjectID
 √ socket entity should "deep equal" passed-in entity

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -89,13 +89,14 @@
     "jshint": "^2.9.3",
     "localstorage-memory": "^1.0.2",
     "mocha": "^3.0.2",
+    "mongodb": "^2.2.29",
```
Comment (marshallswain) [inline package.json]: Is this necessary?

**Comment by marshallswain** [comment]
Oh thank you!  I was going to work on this today, but you just saved my schedule.   One comment about the MongoDB version number and I'll merge this.

File: package.json (hunk 1 above)
Comment (jerfowler) [inline package.json]: Used to generate an ObjectID from a string value. I probably could stub-out similar functionality, though.

Comment (marshallswain) [inline package.json]: Nah.  I'm happy with this.  Thanks!

**Comment by ekryski** [comment]
Yeah nice fixes. Thanks @jerfowler!

**Comment by marshallswain** [comment]
Released as `feathers-authentication@1.2.5`.  

https://github.com/feathersjs/feathers-authentication/blob/master/CHANGELOG.md#v125-2017-06-21

**Pull request merged**